### DISCUSSION TITLE
Trigger setValues on Change w/ less renders

### DISCUSSION THREAD

**Discussion by exrhizo** [react-hook-form/react-hook-form#4533] [Ideas] [answered]
I found that I wanted to have 

```
const field = watch("object_field")
useEffect(() => {
  setValue("object_field_id", field.id)
}, [field])
```

But then that requires a number of renders to accomplish, I wonder about being able to run this more synchronously. 

My current solution is to do this sort of thing with the `onChange` but that isn't as elegant as if I could directly insert callbacks.

**Comment by bluebill1049** [comment, marked as answer]
in v7, you will be able to do the following: 

```
useEffect(() => {
  watch((data) => {
     if(data.test) setValue('test', 'whatever');
  })
})
```

**Reply by exrhizo** [reply]
Appreciation for this library - it's very useful!

I looked for information about this, what do you think about using v7 in production, if not ready yet, when do you expect to consider it released?

**Reply by bluebill1049** [reply]
it will get released in next week, RC version is pretty stable at this point.
